pkg/aggregator/aggregator.go

```diff
@@ -897,10 +908,41 @@ func (agg *BufferedAggregator) run() {
 				}
 			}
 			tlmFlush.Add(1, event.eventType, state)
+		case event := <-agg.contLcycleIn:
+			agg.contlcycleDequeueOnce.Do(func() { go agg.dequeueContainerLifecycleEvents() })
+			aggregatorContainerLifecycleEvents.Add(1)
+			agg.handleContainerLifecycleEvent(event)
 		}
 	}
 }
 
+// dequeueContainerLifecycleEvents consumes buffered container lifecycle events.
+// It is blocking, it should run once and in a separate goroutine.
+func (agg *BufferedAggregator) dequeueContainerLifecycleEvents() {
+	for {
+		select {
+		case event := <-agg.contLcycleBuffer:
+			if err := agg.serializer.SendContainerLifecycleEvent(event.msgs, agg.hostname); err != nil {
+				aggregatorContainerLifecycleEventErrors.Add(1)
+				log.Warnf("Error submitting container lifecycle data: %w", err)
+			}
+		case <-agg.stopChan:
+			return
+		}
+	}
+}
+
+// handleContainerLifecycleEvent forwards container lifecycle events to the buffering channel.
+func (agg *BufferedAggregator) handleContainerLifecycleEvent(event senderContainerLifecycleEvent) {
+	select {
+	case agg.contLcycleBuffer <- event:
+		return
+	default:
+		aggregatorContainerLifecycleEventErrors.Add(1)
+		log.Warn("Container lifecycle events channel is full")
```

Comment:
check: I think you told me not that many events will be processed? I just want to make sure that log message won't be too spammy

Reply:
Yes I think a warning is adequate here, note this is disabled by default and will be tested in prod envs before being rolled out to users. We can always re-evaluate later ofc.